### PR TITLE
[Queue] limit in data queue

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Set max number in queue when sending data to other node.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Comment by jaeyun-jung** [comment]
Related issue: #103

File: src/libnnstreamer-edge/nnstreamer-edge-queue.c [hunk 1]
```diff
@@ -177,12 +199,17 @@ nns_edge_queue_push (nns_edge_queue_h handle, void *data,
...
   nns_edge_lock (q);
-  if (!q->head)
-    q->head = qdata;
-  if (q->tail)
-    q->tail->next = qdata;
-  q->tail = qdata;
-  q->length++;
+  if (q->max_data > 0U && q->length >= q->max_data) {
+    nns_edge_logw ("[Queue] Cannot push new data, max data in queue is %u.",
+        q->max_data);
```
Comment (gichan-jang) [inline src/libnnstreamer-edge/nnstreamer-edge-queue.c:204]: How about setting the buffer to drop out of the old or new buffer in the next patch?

**Review by gichan-jang — APPROVED** [review]
LGTM

File: src/libnnstreamer-edge/nnstreamer-edge-queue.c (hunk 1 above)
Comment (myungjoo) [reply src/libnnstreamer-edge/nnstreamer-edge-queue.c:204]: Yes. It's ok to have a backlog for a leaky queue.

Comment (jaeyun-jung) [reply src/libnnstreamer-edge/nnstreamer-edge-queue.c:204]: I will add new PR for leaky option.

**Pull request merged**